src/main/java/nextstep/subway/line/domain/Line.java

```diff
@@ -53,4 +55,145 @@ public String getColor() {
     public List<Section> getSections() {
         return sections;
     }
+
+    private void addSection(Section section) {
+        this.sections.add(section);
+    }
+
+    public void addSection(Station upStation, Station downStation, int distance) {
+        checkValidationForDuplicationSection(upStation, downStation);
+        checkValidationForValidSection(upStation, downStation, getStations());
+
+        if (isEmptySection(upStation, downStation, distance)) {
+            this.sections.add(new Section(this, upStation, downStation, distance));
+            return;
+        }
+
+        updateSectionStation(upStation, downStation, distance);
+        addSection(new Section(this, upStation, downStation, distance));
+    }
+
+    public void removeStation(Station station) {
+        checkValidationForEmptySections();
+
+        removeStation(getSectionUpLineStation(station), getSectionDownLineStation(station));
```

Comment:
```suggestion
        Section upLineStation = getSectionUpLineStation(station);
        if(upLineStation == null) {
           ... 
           return
        }
        
        removeStation(upLineStation, downLineStation);
```

불확실한 객체를 넘겨주기 보다는 좁은 범위에서 판단하고 확실한 객체만 넘겨준다면,
데이터를 지속적으로 검증해야 하는 부분이 적어 질것 같아요 🤔

Reply:
[Optional 제대로 활용하기](https://www.latera.kr/blog/2019-07-02-effective-optional)

참고해주세요~